### PR TITLE
feat(conductor): add metrics

### PR DESCRIPTION
**Pull request by SuperFluffy**
## Summary
Adds a first small set of metrics to Conductor.

## Background
Conductor was the last service that was entirely without metrics.

## Changes
- Defined, described, and added metrics to conductor.

## Testing
Need to check devnet for the metrics being picked up.

## Metrics
- `astria_conductor_blobs_per_celestia_fetch`
- `astria_conductor_celestia_blob_fetch_error_count`
- `astria_condcutor_blobs_per_celestia_fetch`
- `astria_conductor_sequencer_blocks_metadata_verified_per_celestia_fetch`
- `astria_conductor_sequencer_block_information_reconstructed_per_celestia_fetch`
- `astria_conductor_executed_firm_block_number`
- `astria_conductor_executed_soft_block_number`
- `astria_conductor_transactions_per_executed_block`

### CODE REVIEW

File: crates/astria-conductor/src/celestia/mod.rs
Comment (Fraser999) [inline crates/astria-conductor/src/celestia/mod.rs:538]: I think these either need to be two separate histograms entirely, or we should record `new_blobs.len_header_blobs() + new_blobs.len_rollup_blobs()` in this one.  I don't think Prometheus does anything with the labels, but it also doesn't hurt to add them in case we move away from Prometheus in the future I guess.

**Review by Fraser999 — CHANGES_REQUESTED** [review]
I think there are a couple of comments which could do with getting resolved before this merges.  Apart from that, it'll be good to have metrics in the conductor!

File: crates/astria-conductor/src/celestia/mod.rs
Comment (Fraser999) [reply crates/astria-conductor/src/celestia/mod.rs:538]: After discussion with @SuperFluffy and looking at the code in `metrics` and `metrics-exporter-prometheus` seems like the labels do actually generate separate time series, so this should be fine.

File: crates/astria-conductor/src/celestia/mod.rs
Comment (Fraser999) [reply crates/astria-conductor/src/celestia/mod.rs:572]: As per above follow-on comment.

**Review by Fraser999 — APPROVED** [review]
Approved pending the nitpicks, as my concerns about the histograms seem to be invalid :)

File: crates/astria-conductor/src/metrics_init.rs
Comment (SuperFluffy) [reply crates/astria-conductor/src/metrics_init.rs:68]: Addressed in 1fea99

**Pull request merged**
